src/main/java/ng/abdlquadri/eventbus/handlers/AsyncResult.java

```diff
@@ -0,0 +1,30 @@
+package ng.abdlquadri.eventbus.handlers;

```

Comment:
Yes it requires more thought indeed. And a bit more refactoring, although it is still doable. I'll remove it for this PR.